packages/commonwealth/server/util/roles.ts

```diff
@@ -302,15 +311,26 @@ export async function isAddressPermitted(
 ): Promise<PermissionError | undefined> {
   const roles = await findAllRoles(models, { where: { address_id } }, chain_id);
 
+  // fetch the default allow and deny permissions for the chain
+  const chain = await models.Chain.findOne({ where: { id: chain_id } });
+  if (!chain) {
+    throw new Error('Chain not found');
+  }
+
   // sort roles by roles with highest permissions first
   roles.sort((a) => {
     if (a.permission === 'member') return -1;
     if (a.permission === 'moderator') return 0;
     else return 1;
   });
 
+  const permissionsAllowDeny: Array<{ allow: Permissions; deny: Permissions }> = roles
+
+  // add chain default permissions to beginning of permissions array
```

Comment:
We should try to construct the permissions array more cleanly here, because later on we'll add several more layers of this hierarchy (topics, etc).